renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -689,6 +691,122 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
+    region.imageExtent.height = 1U;
+    region.imageExtent.depth = 1U;
+    region.imageSubresource.baseArrayLayer = m_CallbackInfo.targetSubresource.slice;
+    region.imageSubresource.mipLevel = m_CallbackInfo.targetSubresource.mip;
+    region.imageSubresource.layerCount = 1;
+
+    // TODO: support depth/stencil copy for multi-sampled images.
+    if(p.depthCopy && (m_CallbackInfo.samples != VK_SAMPLE_COUNT_1_BIT))
+      return;
+
+    if(!p.depthCopy)
+    {
+      region.imageSubresource.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+      regions.push_back(region);
+      aspectFlags = VkImageAspectFlags(VK_IMAGE_ASPECT_COLOR_BIT);
+    }
+    else if(p.stencilOnly)
+    {
+      region.imageSubresource.aspectMask = VK_IMAGE_ASPECT_STENCIL_BIT;
+      regions.push_back(region);
+      aspectFlags = VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT;
+    }
+    else
+    {
+      region.imageSubresource.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+      if(IsDepthOnlyFormat(p.srcImageFormat) || IsDepthAndStencilFormat(p.srcImageFormat))
+      {
+        regions.push_back(region);
+        aspectFlags |= VK_IMAGE_ASPECT_DEPTH_BIT;
+      }
+      if(IsStencilFormat(p.srcImageFormat))
+      {
+        region.imageSubresource.aspectMask = VK_IMAGE_ASPECT_STENCIL_BIT;
+        region.bufferOffset = offset + 4;
+        regions.push_back(region);
+        aspectFlags |= VK_IMAGE_ASPECT_STENCIL_BIT;
+      }
+    }
+
+    VkImage cmdCopySource = p.srcImage;
+    VkImageLayout cmdCopySourceLayout = p.srcImageLayout;
+    uint32_t baseArrayLayer = m_CallbackInfo.targetSubresource.slice;
+
+    // For multi-sampled images can't call vkCmdCopyImageToBuffer directly,
+    // copy using a compute shader into a staging image first.
+    if(m_CallbackInfo.samples != VK_SAMPLE_COUNT_1_BIT)
+    {
+      regions[0].imageSubresource.baseArrayLayer = 0;
+      VkImageMemoryBarrier barrier = {VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER,
+                                      NULL,
+                                      VK_ACCESS_SHADER_WRITE_BIT |
+                                          VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT |
+                                          VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT |
+                                          VK_ACCESS_TRANSFER_WRITE_BIT | VK_ACCESS_MEMORY_WRITE_BIT,
+                                      VK_ACCESS_SHADER_READ_BIT,
+                                      p.srcImageLayout,
+                                      VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL,
+                                      VK_QUEUE_FAMILY_IGNORED,
+                                      VK_QUEUE_FAMILY_IGNORED,
+                                      Unwrap(p.srcImage),
+                                      {aspectFlags, 0, 1, baseArrayLayer, 1}};
+
+      DoPipelineBarrier(cmd, 1, &barrier);
+      m_pDriver->GetDebugManager()->CopyTex2DMSPixel(
+          cmd, m_CallbackInfo.extent, 0 /* m_CallbackInfo.targetSubresource.slice */,
```

Comment:
Why is the slice commented out here? Are MSAA arrays yet to be implemented?

Reply:
I don't understand the follow-up change you made - the slice is now hardcoded as 0 and there's no parameter at all. Is there something else that needs to be done to support MSAA arrays or could they be supported if the parameter is passed all the way through from here?

Reply:
Sorry I forgot to reply last night.
When I create the target image view (the source image view here for the image for which pixel history is requested) I select the slice/layer already there. So here there is only one to choose from. I added a comment where the image view is creaetd.

Reply:
Ahh I see, that makes sense thanks!